drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -855,6 +856,37 @@ static void nrf5_config_mac_keys(struct ieee802154_key *mac_keys)
 }
 #endif /* CONFIG_NRF_802154_ENCRYPTION */
 
+#if defined(CONFIG_IEEE802154_CSL_ENDPOINT)
+
+static void nrf5_receive_at(const struct ieee802154_config *config)
+{
+	bool primary_cancel = nrf_802154_receive_at_scheduled_cancel(DRX_SLOT_RX_PRIMARY);
+	bool secondary_cancel = nrf_802154_receive_at_scheduled_cancel(DRX_SLOT_RX_SECONDARY);
+	if (!primary_cancel && !secondary_cancel) {
+		/* Edge case: In between the primary and secondary slot cancellation
+		 *            the secondary RX slot might have started. Due to ordering
+		 *            of the above two cancellations we can assume that the primary
+		 *            slot is already free.
+		 */
+		primary_cancel = true;
+	}
+
+	int slot_id = primary_cancel ? DRX_SLOT_RX_PRIMARY : DRX_SLOT_RX_SECONDARY;
+
+	/* Note that even if the nrf_802154_receive_at function is not called in time
+	 * (for example due to the call being blocked by higher priority threads) and
+	 * the delayed reception window is not scheduled, the CSL phase will still be
+	 * calculated as if the following reception windows were at times
+	 * anchor_time + n * csl_period. The previously set
+	 * anchor_time will be used for calculations.
+	 */
+	nrf_802154_receive_at(config->rx_slot.start / NSEC_PER_USEC,
+			      config->rx_slot.duration / NSEC_PER_USEC,
+			      config->rx_slot.channel, slot_id);
```

Comment:
`nrf_802154_receive_at` returns `bool`. What with the result?

Reply:
The previous implementation didn't use the return result. I kept the previous approach.
If proper error handling is desired, I would suggest that it's done more holistically in the entire file.

Reply:
No I'm not forcing a change.,`(void)nrf_802154_receive_at` would be enough, to possibly satisfy some lint.